source/patterns/@aws-solutions-constructs/aws-events-rule-sns/test/events-rule-sns-topic.test.ts

```diff
@@ -0,0 +1,146 @@
+            Statement:  [
+               {
+                Action:  [
+                  "SNS:Publish",
+                  "SNS:RemovePermission",
+                  "SNS:SetTopicAttributes",
+                  "SNS:DeleteTopic",
+                  "SNS:ListSubscriptionsByTopic",
+                  "SNS:GetTopicAttributes",
+                  "SNS:Receive",
+                  "SNS:AddPermission",
+                  "SNS:Subscribe",
+                ],
+                Condition:  {
+                  "StringEquals":  {
+                    "AWS:SourceOwner": {
+                      "Ref": "AWS::AccountId"
+                    },
+                  },
+                },
+                Effect: "Allow",
+                Principal:  {
+                  "AWS":  {
+                    "Fn::Join":  [
+                      "",
+                       [
+                        "arn:",
+                         {
+                          "Ref": "AWS::Partition",
+                        },
+                        ":iam::",
+                        {
+                          "Ref": "AWS::AccountId",
+                        },
+                        ":root",
+                      ],
+                    ],
+                  },
+                },
+                Resource:  {
+                  "Ref": "testeventsrulesnsSnsTopicCEB51DAD",
+                },
+                Sid: "TopicOwnerOnlyAccess",
+              },
+               {
+                Action:  [
+                  "SNS:Publish",
+                  "SNS:RemovePermission",
+                  "SNS:SetTopicAttributes",
+                  "SNS:DeleteTopic",
+                  "SNS:ListSubscriptionsByTopic",
+                  "SNS:GetTopicAttributes",
+                  "SNS:Receive",
+                  "SNS:AddPermission",
+                  "SNS:Subscribe",
+                ],
+                Condition:  {
+                  "Bool":  {
+                    "aws:SecureTransport": "false",
+                  },
+                },
+                Effect: "Deny",
+                Principal: "*",
+                Resource:  {
+                  "Ref": "testeventsrulesnsSnsTopicCEB51DAD",
+                },
+                Sid: "HttpsOnly",
+              },
+               {
+                Action: "sns:Publish",
+                Effect: "Allow",
+                Principal: {
+                  "AWS": {
+                    "Fn::GetAtt": [
+                      "testeventsrulesnsEventsRule5F1C01CC",
+                      "Arn",
+                    ],
+                  },
+                },
+                Resource:  {
+                  "Ref": "testeventsrulesnsSnsTopicCEB51DAD",
+                },
+                Sid: "2",
+              },
+            ],
+            Version: "2012-10-17",
+          },
+          Topics:  [
+             {
+              "Ref": "testeventsrulesnsSnsTopicCEB51DAD",
+            },
+          ],
+        },
+    )
+})
```

Comment:
Suggest add a test case to assert the events rule properties, for reference see [here](https://github.com/awslabs/aws-solutions-constructs/blob/master/source/patterns/%40aws-solutions-constructs/aws-events-rule-lambda/test/events-rule-lambda.test.ts#L142)